_transcripts/013-rust-1.41.0.md

```diff
@@ -0,0 +1,801 @@
+apply a bunch of optimizations which takes a bunch of time. And what this
+will let you do is, say, for this dependency or this set of dependencies,
+like build scripts or `proc_macro`, as has been mentioned, I don't want to
+do optimizations because they don't matter. So even if I compile in release
+mode, I don't want to optimize those libraries. And this might buy you a
+bunch of— basically, it might give you `cargo build` performance for some
+of your dependencies, even though you're building with release, without any
+real cost to your final binary.
+
+The other use case for this, which is what got me a bit excited, is to do
+it the other way around, where, imagine that you have some library and you
+depend on a library that, just like, is extremely slow in the default debug
+mode. And so if you run your test suite and you build this dependency in
+debug mode, your test suite just takes forever. Examples of this are things
+that do compression, encryption, dealing with image files, that sort of
+stuff where you really just always want to compile them in release mode,
+even if you're doing a debug build, because otherwise your debug build is
+just going to be horribly slow to run. And so here you can now do that with
+these profile overrides as well. You can say, like `profile.dev`, which is,
+like, the default development build, `.package.compression-handling-library`,
+and set `opt-level = 3`. And now it will be built to be fast and then all
+the other code will still be built quickly and in debug mode. And so this is
+a really nice way to sort of customize your build, so that the things that
+need to be fast are built slowly so that they're fast, and the things that
+can be slow are built quickly and are slow to run.
+
+__Ben__: So I think that's all for our release notes this time.
+
+__Jon__: I had one more, actually, that I found while scrolling through the
+infinite list of changes, which is, you can now arbitrarily nest receiver
+types in the self position. Now this is a little bit weird. Some of you might
+be aware that, in addition, to be able to write methods for impl blocks to,
+say, like `&self` or `&mut self` or just take ownership of `self`, you can
+also write `self:` and then some type that dereferences to self. And the prime
+example of this is `Pin`, that some of you might be aware of, right?  You can
+write `fn poll(self: Pin<&mut Self>)`. And previously you could do this with
+other types as well. You could implement a method for your type, that's like
+`self: Box<Self>`. So that method would only be callable if the caller has a
+`Box` of your type. This might be handy, where in certain cases, you might
+want to have this restriction, but previously you could only go one type deep
+so you could say `self: Arc<Self>`, `self:Box<Self>`. Now you can nest these
+arbitrarily; you can do things like `self: Arc<Box<Rc<RefCell<Self>>>>`.
+(??? _this area should be reviewed_, ~38:21)
```

Comment:
This also needs review.  There was a lot of Rust syntax being spelled out, and I think I got it all, but it would help to have this double-checked.

Reply:
One these reviews are complete, I would like to remove the `(??? _needs review)` markers before merging.

Reply:
Yes, this looks accurate to me!